spec/AsComponent.coffee

```diff
@@ -0,0 +1,252 @@
+        loader.load 'ascomponent/sync-one', (err, instance) ->
+          return done err if err
+          ins = new noflo.internalSocket.createSocket()
+          out = new noflo.internalSocket.createSocket()
+          error = new noflo.internalSocket.createSocket()
+          instance.inPorts.hello.attach ins
+          instance.outPorts.out.attach out
+          instance.outPorts.error.attach error
+          received = []
+          expected = [
+            'openBracket a'
+            'data Hello Foo'
+            'data Hello Bar'
+            'data Hello Baz'
+            'closeBracket a'
+          ]
+          error.once 'data', (data) ->
+            done data
+          out.on 'ip', (ip) ->
+            received.push "#{ip.type} #{ip.data}"
+            return unless received.length is expected.length
+            chai.expect(received).to.eql expected
+            done()
+          ins.post new noflo.IP 'openBracket', 'a'
+          ins.post new noflo.IP 'data', 'Foo'
+          ins.post new noflo.IP 'data', 'Bar'
+          ins.post new noflo.IP 'data', 'Baz'
+          ins.post new noflo.IP 'closeBracket', 'a'
+    describe 'with a thrown exception', ->
+      func = (hello) ->
+        throw new Error "Hello #{hello}"
+      it 'should be possible to componentize', (done) ->
+        component = -> noflo.asComponent func
+        loader.registerComponent 'ascomponent', 'sync-throw', component, done
+      it 'should send to ERROR port', (done) ->
+        wrapped = noflo.asCallback 'ascomponent/sync-throw',
+          loader: loader
+        wrapped 'Error', (err) ->
+          chai.expect(err).to.be.an 'error'
+          chai.expect(err.message).to.equal 'Hello Error'
+          done()
+  describe 'with a synchronous function taking a multiple parameters', ->
+    describe 'with returned value', ->
+      func = (greeting, name) ->
+        return "#{greeting} #{name}"
+      it 'should be possible to componentize', (done) ->
+        component = -> noflo.asComponent func
+        loader.registerComponent 'ascomponent', 'sync-two', component, done
+      it 'should be loadable', (done) ->
+        loader.load 'ascomponent/sync-two', done
+      it 'should contain correct ports', (done) ->
+        loader.load 'ascomponent/sync-two', (err, instance) ->
+          return done err if err
+          chai.expect(Object.keys(instance.inPorts.ports)).to.eql ['greeting', 'name']
+          chai.expect(Object.keys(instance.outPorts.ports)).to.eql ['out', 'error']
+          done()
+      it 'should send to OUT port', (done) ->
+        wrapped = noflo.asCallback 'ascomponent/sync-two',
+          loader: loader
+        wrapped
+          greeting: 'Hei'
+          name: 'Maailma'
+        , (err, res) ->
+          return done err if err
+          chai.expect(res).to.eql
+            out: 'Hei Maailma'
+          done()
+    describe 'with a default value', ->
+      before ->
+        @skip() if isBrowser
```

Comment:
Worthy of a comment to explain that this is due to missing ES6 support. So we can remove when we can assume that